lib/src/event.dart

```diff
@@ -356,6 +356,20 @@ class Event extends MatrixEvent {
 
     await room.client.database?.removeEvent(eventId, room.id);
     room.client.onCancelSendEvent.add(eventId);
+
+    if (room.lastEvent?.eventId == eventId) {
+      Logs().v('cancel send called on lastEvent, setting to null');
+      room.lastEvent = null;
```

Comment:
Again this would move the room to the bottom of a chat list of potential 1000 rooms, leading to that the room magically disappears

Reply:
Why not handling it like WhatsApp? There the last event then is displayed as redacted. Just redact the event without doing a network call

Reply:
updated